src/mlpack/tests/init_rules_test.cpp

```diff
@@ -126,6 +129,35 @@ BOOST_AUTO_TEST_CASE(OivsInitTest)
   BOOST_REQUIRE_EQUAL(1, 1);
 }
 
+BOOST_AUTO_TEST_CASE(XavierInitTest)
+{
+  arma::mat weights;
+  //RandomSeed(21);
+  XavierInit<XavierUniform> xavierUniform;
+  XavierInit<XavierNormal> xavierNormal;
+  xavierUniform.Initialize(weights, 100, 100);
+  bool b = arma::all(arma::vectorise(weights) >= -(1e-2) || arma::vectorise(weights) <= (1e-2));
+  BOOST_REQUIRE_EQUAL(b, 1);
+  xavierNormal.Initialize(weights, 100, 100);
+  //std::cout << weights << std::endl;
+  auto ret = Shapiro(weights, 0.05);
```

Comment:
Maybe we could discuss what other tests we would need for the Shapiro-Wilk test. Because I have used the train method from GaussianDistribution before but it does converge where I mean even to come close to original mean and variance util a high amount of iterations